gwcs/coordinate_frames.py

```diff
@@ -286,12 +286,15 @@ def axis_physical_types(self):
         return self._axis_physical_types
 
     @property
-    def _world_axis_object_components(self):
-        raise NotImplementedError(f"This method is not implemented for {type(self)}")
+    def _world_axis_object_classes(self):
+        return {self._axes_type[0]: (
+            u.Quantity,
+            (),
+            {'unit': self.unit[0]})}
 
     @property
-    def _world_axis_object_classes(self):
-        raise NotImplementedError(f"This method is not implemented for {type(self)}")
+    def _world_axis_object_components(self):
+        return [(self._axes_type[0], 0, 'value')]
```

Comment:
I can see this working in the case of `Time` but what about any other axes types

```
slit = cf.CoordinateFrame(naxes=1, axes_type='SPATIAL', axes_order=(0,),  axes_names=('slit',), unit=u.pix)
```

Reply:
This implementation makes the assumption that the coordinates of the BaseFrame are always `u.Quantity` objects, which given unit and the implementation of the coordinates method in the base frame holds up. The `self._axes_type[0]` is used as a shared ID across components and classes.

This section of the PR has nothing to do with the TemporalFrame, that implements it's own version of this method. I am using this method to generate a 3D grid of coordinates in distance units (i.e. for a simulation).

Reply:
OK. Can you add tests for these two methods using quantities.